c_emulator/riscv_sim.cpp

```diff
@@ -128,7 +128,8 @@ static struct option options[] = {
 #endif
     {"print-device-tree",              no_argument,       0, OPT_PRINT_DTS      },
     {"print-isa-string",               no_argument,       0, OPT_PRINT_ISA      },
-    {0,                                0,                 0, 0                  }
+    {0,                                0,                 0, 0                  },
+    {"dynamic",                        required_argument, 0, 'd'                }
```

Comment:
Uh how does this option work if it's after the terminator? Presumably the short option works because that's separate, but the long option doesn't (the 'd' here is to make the long option report the same result as for the short one parsed by the alphabet soup string).

Reply:
so how could I fix it(?

Reply:
Well you could start by testing this option and observing that it doesn't work. Then fix it by swapping lines 131 and 132.

Reply:
everything looks good.